src/main/java/io/github/oliviercailloux/y2018/apartments/utils/JsonConvert.java

```diff
@@ -136,17 +95,19 @@ public static Apartment jsonToApartment(Path jsonPath) throws IOException {
 	public static List<Apartment> jsonToApartments(Path jsonPath) throws IOException {
 		String jsonString = Files.readString(jsonPath);
 		List<Apartment.Builder> apartmentsBuild;
-		List<Apartment> apartments = new ArrayList<Apartment>();
+		List<Apartment> apartments = new ArrayList<>();
 		LOGGER.info("Create ArrayList of Apartment");
 
-		Jsonb jsonb = JsonbBuilder.create();
-		LOGGER.info("Create Json builder");
-
-		apartmentsBuild = jsonb.fromJson(jsonString, new ArrayList<Apartment.Builder>() {
-		}.getClass().getGenericSuperclass());
+		try (Jsonb jsonb = JsonbBuilder.create()) {
+			LOGGER.info("Create Json builder");
+			apartmentsBuild = jsonb.fromJson(jsonString, new ArrayList<Apartment.Builder>() {
```

Comment:
Il devrait j’imagine être possible, et serait plus élégant, de demander à Json-B de créer directement une liste d’appartements plutôt qu’une liste de builders (dont on ne fait rien d’autre que la convertir en liste d’appartements). Mais je n’ai pas vérifié. FUTURE (indiquez dans l’issue créée à cet effet si vous avez fait ce choix pour une raison particulière.)

Reply:
Nous avions déjà étudié la possibilité de créer directement une liste d'Apartment, mais en raison du caractère privé des attributs de la classes et des setters, nous avions abandonné cette idée. Nous réaliserons une enquête dans l'itération 3 pour voir s'il existe une solution. https://github.com/av1m/Apartments/issues/20